osu.Game/Online/API/Requests/Responses/APIFriend.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Newtonsoft.Json;
+using osu.Game.Users;
+
+namespace osu.Game.Online.API.Requests.Responses
+{
+    public class APIFriend : User
+    {
+        [JsonProperty(@"current_mode_rank")]
+        public int? CurrentModeRank;
+    }
```

Comment:
This inheritance feels gratuitous. Naming aside (what is an `APIFriend`?), as far as I can see web [does not return a full user here, but instead a trimmed structure with only partial fields filled out](https://github.com/ppy/osu-web/blob/4687a1e8a5c132af3ccfc63a2b90d239ec395d71/app/Http/Controllers/FriendsController.php#L46). If you try to pass this data around pretending it's a `User` it'll break somewhere where an actual `User` with the full data is required (in short, this breaks the Liskov substitution principle).

Reply:
Well, we have many cases like this in-game (where web does not return a full user, but we are using one)

Reply:
Should I just add this property to `User`?

Reply:
we cannot use subclassing on models. it is not supported by realm and not the direction we want to take things.